app/src/main/java/com/simonramstedt/yoke/YokeActivity.java

```diff
@@ -183,59 +216,69 @@ protected void onPreExecute() {
+                    int count;
+                    publishProgress(0L, INDETERMINATE);
                     while ((count = input.read(data)) != -1) {
-                        cumulativeBytes += count;
                         output.write(data, 0, count);
-                        publishProgress(cumulativeBytes, totalBytes);
                     }
+                    BufferedReader inputBR = new BufferedReader(
+                        new FileReader(futureManifestPath)
+                    );
+                    String line = "";
+                    while ((line = inputBR.readLine()) != null) {
+                        manifestSB.append(line).append("\n");
+                    }
+                    JSONObject manifestJSON = new JSONObject(manifestSB.toString());
+
+                    long totalBytes = manifestJSON.optLong("size", INDETERMINATE);
+                    if (totalBytes != INDETERMINATE)
+                        publishProgress(0L, DETERMINATE);
+                    JSONArray entries = manifestJSON.getJSONArray("folders");
+                    for (int i = 0, length = entries.length(); i < length; i++) {
+                        File newFolder = new File(futureJoypadPath, entries.getString(i));
+                        log(String.format(res.getString(R.string.log_creating_folder), newFolder.getAbsolutePath()));
+                        if (!newFolder.mkdirs()) {
+                            throw new IOException(String.format(
+                                res.getString(R.string.error_could_not_create_folder), newFolder.getAbsolutePath()
+                            ));
+                        }
+                    }
+
+                    entries = manifestJSON.getJSONArray("files");
+                    long cumulativeBytes = 0;
+                    for (int i = 0, length=entries.length(); i < length; i++) {
+                        File newFile = new File(futureJoypadPath, entries.getString(i));
+                        log(String.format(res.getString(R.string.log_downloading_file), newFile.getAbsolutePath()));
+                        input = new BufferedInputStream(new URL(f_url[0] + entries.getString(i)).openStream(), 8192);
+                        output = new FileOutputStream(newFile);
+
+                        while ((count = input.read(data)) != -1) {
+                            cumulativeBytes += count;
+                            output.write(data, 0, count);
+                            publishProgress(cumulativeBytes, totalBytes);
+                        }
 
-                    output.flush();
-                    output.close();
-                    input.close();
+                        output.flush();
+                        output.close();
+                        input.close();
+                    }
+                    return UPDATE_SUCCESS;
+                } else {
+                    // download just one file
```

Comment:
This comment confused me a bit at first, as from what I understand, the file is not saved anywhere, and this conditional branch purpose is to just check if the server is online (and serves the right file)?

Reply:
Yes. Exactly that.

Maybe in the future we could actually parse the file to compare versions or something, but for now, the file is just discarded.